dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/LuceneQueryDateTimeFormatter.java

```diff
@@ -0,0 +1,253 @@
+            token = tokens[pos];
+            if (token.matches("\\S+\\.\\S+:\\S*")) {
+                clauses.add(token);
+            } else if (token.matches("\\d+:\\S*")) {
+                clauses.set(clauses.size() - 1, clauses.get(clauses.size() - 1) + " " + token);
+            } else if (token.matches("\\[\\S*")) {
+                clauses.set(clauses.size() - 1, clauses.get(clauses.size() - 1) + token);
+            } else if (token.matches("TO") || token.toLowerCase().matches("am") || token.toLowerCase().matches("pm")) {
+                clauses.set(clauses.size() - 1, clauses.get(clauses.size() - 1) + " " + token);
+            } else if (token.matches("\\S*\\]")) {
+                clauses.set(clauses.size() - 1, clauses.get(clauses.size() - 1) + " " + token);
+            } else if (token.matches("\\d{1,2}/\\d{1,2}/\\d{4}")) {
+                clauses.set(clauses.size() - 1, clauses.get(clauses.size() - 1) + " " + token);
+            } else {
+                clauses.add(token);
+            }
+        }
+
+        //DOTCMS - 4127
+        List<Field> dateFields = new ArrayList<Field>();
+        String tempStructureVarName;
+        Structure tempStructure;
+
+        for (String clause: clauses) {
+
+            // getting structure names from query
+            if(clause.indexOf('.') >= 0 && (clause.indexOf('.') < clause.indexOf(':'))){
+
+                tempStructureVarName = clause.substring(0, clause.indexOf('.'));
+                tempStructure = CacheLocator.getContentTypeCache().getStructureByVelocityVarName(tempStructureVarName);
+
+                if (UtilMethods.isSet(tempStructure) && UtilMethods.isSet(tempStructure.getVelocityVarName())) {
+                    List<Field> tempStructureFields = new ArrayList<>(FieldsCache
+                            .getFieldsByStructureVariableName(
+                                    tempStructure.getVelocityVarName()));
+
+                    for (int pos = 0; pos < tempStructureFields.size(); ) {
+
+                        if (tempStructureFields.get(pos).getFieldType()
+                                .equals(Field.FieldType.DATE_TIME.toString()) ||
+                                tempStructureFields.get(pos).getFieldType()
+                                        .equals(Field.FieldType.DATE.toString()) ||
+                                tempStructureFields.get(pos).getFieldType()
+                                        .equals(Field.FieldType.TIME.toString())) {
+                            ++pos;
+                        } else {
+                            tempStructureFields.remove(pos);
+                        }
+
+                    }
+
+                    dateFields.addAll(tempStructureFields);
+                }
+            }
+        }
+
+        String replace;
+        List<RegExMatch> matches;
+        String structureVarName;
+        for (String clause: clauses) {
+            for (Field field: dateFields) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'field' could be declared final](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4545506)